alacritty/src/renderer/solidrect.rs

```diff
@@ -0,0 +1,200 @@
+use alacritty_terminal::term::SizeInfo;
+
+use crate::gl;
+use crate::gl::types::*;
+use crate::renderer::rects::RenderRect;
+use crate::renderer::RectShaderProgram;
+
+use std::mem::size_of;
+use std::ptr;
+
+const MAX_U16_INDICES: usize = 65536;
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+struct Rgba {
+    r: u8,
+    g: u8,
+    b: u8,
+    a: u8,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+struct Vertex {
+    // TODO these can certainly be i16
+    x: f32,
+    y: f32,
```

Comment:
I think macOS doesn't like i16 in general, so I'd go for f32, though, we can test  that it works on macOS.

Reply:
There's also a problem where rect coordinates come with non-integer pixel coordinates. I looked at it briefly and it has to do with non-integer thickness, did not investigate further.

Reply:
That sounds strange, but I think possible, it's not like you have a half a pixel on the screen.

Reply:
> it's not like you have a half a pixel on the screen.

I mean technically, subpixel positioning is a thing sooo.. :)

Reply:
Yeah, vertex coordinates have subpixel precision even without fancy multisampling stuff. If they hadn't, we'd all be looking at jittery 3D graphics from 90s.